stacks/ring.go

```diff
@@ -105,11 +109,12 @@ func (r *ring) onReadEnd() {
 		r.off = 0 // Wrap around.
 	}
 	if r.off == r.end {
-		r.Reset() // We read everything, reset.
+		r.Reset() // We read everything, reset.  - why reset - ever ?
```

Comment:
Reduces split buffer reads, which invokes two memcpy instead of only one.

Reply:
Ahh !